src/BinSkim.Driver/RoslynAnalyzer/ILOperations.cs

```diff
@@ -0,0 +1,485 @@
+        public string Spelling => ConstantValue.ToString();
+    }
+
+    internal sealed class AddressOfExpression : Expression, IAddressOfExpression
+    {
+        public AddressOfExpression(Compilation compilation, IReferenceExpression addressed)
+        {
+            Addressed = addressed;
+            ResultType = compilation.CreatePointerTypeSymbol(addressed.ResultType);
+        }
+
+        public IReferenceExpression Addressed { get; }
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.AddressOfExpression;
+    }
+
+    internal sealed class ArrayCreationExpression : Expression, IArrayCreationExpression
+    {
+        public ArrayCreationExpression(Compilation compilation, ITypeSymbol elementType, ImmutableArray<IExpression> dimensionSizes)
+        {
+            ElementType = elementType;
+            DimensionSizes = dimensionSizes;
+            ResultType = compilation.CreateArrayTypeSymbol(elementType, dimensionSizes.Length);
+        }
+
+        public ITypeSymbol ElementType { get; }
+        public ImmutableArray<IExpression> DimensionSizes { get; }
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.ArrayCreationExpression;
+
+        IArrayInitializer IArrayCreationExpression.ElementValues => null;
+    }
+
+    internal sealed class AssignmentExpression : Expression, IAssignmentExpression
+    {
+        public AssignmentExpression(IReferenceExpression target, IExpression value)
+        {
+            Target = target;
+            Value = value;
+        }
+
+        public IReferenceExpression Target { get; }
+        public IExpression Value { get; }
+
+        public override OperationKind Kind => OperationKind.AssignmentExpression;
+        public override ITypeSymbol ResultType => Target.ResultType;
+    }
+
+    internal abstract class Statement : Operation, IStatement
+    {
+    }
+
+    internal sealed class ExpressionStatement : Statement, IExpressionStatement
+    {
+        public ExpressionStatement(IExpression expression)
+        {
+            Expression = expression;
+        }
+
+        public IExpression Expression { get; }
+        public override OperationKind Kind => OperationKind.ExpressionStatement;
+    }
+
+    internal class BlockStatement : Statement, IBlockStatement
+    {
+        public BlockStatement(ImmutableArray<ILocalSymbol> locals, ImmutableArray<IStatement> statements)
+        {
+            Locals = locals;
+            Statements = statements;
+        }
+
+        public ImmutableArray<ILocalSymbol> Locals { get; }
+        public ImmutableArray<IStatement> Statements { get; }
+        public override OperationKind Kind => OperationKind.BlockStatement;
+    }
+
+    // FEEDBACK: Why is there both ILabelStatement and ILabeledStatement? Not sure which to use.
```

Comment:
I'll have to do a little more thinking to be sure that this works out OK. It would turn a C# labeled statement into two statements, which might introduce some awkward situations.